tests/unit/generator_test.py

```diff
@@ -79,32 +83,49 @@ class TestType(mty.Type):
         Generator().generate(Model([TestType("P::T")]), Integration(), tmp_path)
 
 
+@pytest.mark.parametrize(
+    "debug, debug_expected",
+    [
+        (Debug.NONE, set()),
+        (Debug.BUILTIN, set()),
+        (Debug.EXTERNAL, {"rflx_debug.ads"}),
+    ],
+)
 @pytest.mark.parametrize(
     "prefix, library_files, top_level_package, expected",
     [
-        ("RFLX", True, True, ["rflx.ads", *[f"rflx-{f}" for f in const.LIBRARY_FILES]]),
-        ("RFLX", True, False, [f"rflx-{f}" for f in const.LIBRARY_FILES]),
-        ("RFLX", False, True, ["rflx.ads"]),
-        ("RFLX", False, False, []),
-        ("", True, True, const.LIBRARY_FILES),
-        ("", True, False, const.LIBRARY_FILES),
-        ("", False, True, []),
-        ("", False, False, []),
+        ("RFLX", True, True, {"rflx.ads"} | {f"rflx-{f}" for f in const.LIBRARY_FILES}),
+        ("RFLX", True, False, {f"rflx-{f}" for f in const.LIBRARY_FILES}),
+        ("RFLX", False, True, {"rflx.ads"}),
+        ("RFLX", False, False, set()),
+        ("", True, True, set(const.LIBRARY_FILES)),
+        ("", True, False, set(const.LIBRARY_FILES)),
+        ("", False, True, set()),
+        ("", False, False, set()),
     ],
 )
+# pylint: disable-next = too-many-arguments
 def test_generate(
-    prefix: str, library_files: bool, top_level_package: bool, expected: list[str], tmp_path: Path
+    debug: Debug,
+    debug_expected: Set[str],
+    prefix: str,
+    library_files: bool,
+    top_level_package: bool,
+    expected: Set[str],
+    tmp_path: Path,
 ) -> None:
-    Generator(prefix, reproducible=True).generate(
+    Generator(prefix, reproducible=True, debug=debug).generate(
         Model(), Integration(), tmp_path, library_files, top_level_package
     )
-    for filename in expected:
-        if prefix:
-            assert (tmp_path / filename).read_text() == (
-                GENERATED_DIR / filename
-            ).read_text(), filename
-        else:
-            assert (tmp_path / filename).exists()
+    present = {f.name for f in tmp_path.glob("*.ad?")}
+    assert present == expected | (
+        {(f"{file_name(prefix)}-{f}" if prefix else f) for f in debug_expected}
+        if library_files
+        else set()
+    )
+
+    if prefix and not debug:
+        assert all((tmp_path / f).read_text() == (GENERATED_DIR / f).read_text() for f in present)
```

Comment:
That's true - changed. Looks like we need coverage analysis for our tests - in theory no part of the tests should not be executed...

Reply:
We could enable the coverage analysis for the actual test code (by adding `--cov=tests`), but as we exclude some tests during coverage analysis, we will never get 100% test coverage. It could be possible if we restructure the `tests` directory, so that compilation and verification tests are in separate directories.

Reply:
Doing that seems worthwhile. But not today: #1190